File: modules/base/examples/flags.py
## \example base/flags.py
# A simple example of how to use IMP flags support in a python
# executable. Run with "--help" to see all available flags.

import IMP.base
import sys

IMP.base.add_string_flag("hello", "Hello world", "How to say hello")

IMP.base.setup_from_argv(sys.argv, "An example for using flags in python.")

if IMP.base.get_string_flag("hello") == "":
    IMP.base.write_help()
    exit(1)

print IMP.base.get_string_flag("hello")

File: modules/base/examples/flags.cpp
/** \example base/flags.cpp
    Show simple usage of the command line flags support in IMP.base in a
    C++ executable. Run with "--help" to see all available flags.
*/
#include <IMP/base/flags.h>
#include <IMP/base/Flag.h>

int main(int argc, char *argv[]) {
  IMP::base::Flag<std::string> hello("hello", "How to say hello.",
                                     "Hello world");

  IMP::base::setup_from_argv(argc, argv, "Example on C++ use of flags.");

  std::cout << hello << std::endl;

  return 0;
}
